sdks/python/apache_beam/transforms/enrichment_handlers/vertex_ai_feature_store.py

```diff
@@ -0,0 +1,289 @@
+
+  **NOTE:** The default severity to report exceptions is logging a warning. For
+    this handler, Vertex AI client returns the same exception
+    `Requested entity was not found` even though the feature store doesn't
+    exist. So make sure the feature store instance exists or set
+    `exception_level` as `ExceptionLevel.RAISE`.
+  """
+  def __init__(
+      self,
+      project: str,
+      location: str,
+      api_endpoint: str,
+      feature_store_name: str,
+      feature_view_name: str,
+      row_key: str,
+      *,
+      exception_level: ExceptionLevel = ExceptionLevel.WARN,
+      **kwargs,
+  ):
+    """Initializes an instance of `VertexAIFeatureStoreEnrichmentHandler`.
+
+    Args:
+      project (str): The GCP project-id for the Vertex AI Feature Store.
+      location (str): The region for the Vertex AI Feature Store.
+      api_endpoint (str): The API endpoint for the Vertex AI Feature Store.
+      feature_store_name (str): The name of the Vertex AI Feature Store.
+      feature_view_name (str): The name of the feature view within the
+        Feature Store.
+      row_key (str): The row key field name containing the unique id
+        for the feature values.
+      exception_level: a `enum.Enum` value from
+        `apache_beam.transforms.enrichment_handlers.utils.ExceptionLevel`
+        to set the level when an empty row is returned from the BigTable query.
+        Defaults to `ExceptionLevel.WARN`.
+      kwargs: Optional keyword arguments to configure the
+        `aiplatform.gapic.FeatureOnlineStoreServiceClient`. When using
+        `kwargs`, the `api_endpoint` param will be overridden with this config
+        if multiple values are found for `client_options`.
+    """
+    self.project = project
+    self.location = location
+    self.api_endpoint = api_endpoint
+    self.feature_store_name = feature_store_name
+    self.feature_view_name = feature_view_name
+    self.row_key = row_key
+    self.exception_level = exception_level
+    self.kwargs = kwargs if kwargs else {}
+
+  def __enter__(self):
+    """Connect with the Vertex AI Feature Store."""
+    if 'client_options' in self.kwargs:
+      self.client = aiplatform.gapic.FeatureOnlineStoreServiceClient(
+          **self.kwargs)
+    else:
+      self.client = aiplatform.gapic.FeatureOnlineStoreServiceClient(
+          client_options={"api_endpoint": self.api_endpoint}, **self.kwargs)
+    self.feature_view_path = self.client.feature_view_path(
+        self.project,
+        self.location,
+        self.feature_store_name,
+        self.feature_view_name)
+
+  def __call__(self, request: beam.Row, *args, **kwargs):
+    """Fetches feature value for an entity-id from Vertex AI Feature Store.
+
+    Args:
+      request: the input `beam.Row` to enrich.
+    """
+    try:
+      entity_id = request._asdict()[self.row_key]
+    except KeyError:
+      raise ValueError(
+          "no entry found for row_key %s in input row" % self.row_key)
```

Comment:
Could we add a little detail to this error (e.g. `All Enrichment requests must have a row key to join on. This will be used as the FeatureViewDataKey`).